app/services/fact_importer.rb

```diff
@@ -14,6 +14,15 @@ def self.register_fact_importer(key, klass)
     importers[key.to_sym] = klass
   end
 
+  def self.fact_features
+    importers.map { |_type, importer| importer.authorized_smart_proxy_features }.compact.flatten.sort.uniq
```

Comment:
Oh sorry, that's a bashism - you always have to sort before running uniq, it's not smart enough otherwise. Ruby is. I don't really care about the order. :-)